File: mozn-dashboard/server/internal/model/activity.go
package model

// ActivityCategory classifies an audit-log entry.
type ActivityCategory string

const (
	CategoryAlert     ActivityCategory = "alert"
	CategoryThreshold ActivityCategory = "threshold"
	CategoryStation   ActivityCategory = "station"
	CategoryUser      ActivityCategory = "user"
	CategoryAuth      ActivityCategory = "auth"
)

// ActivityRow is one audit-log entry.
type ActivityRow struct {
	ID       string           `json:"id"`
	Time     string           `json:"time"`
	Actor    string           `json:"actor"`
	Initials string           `json:"initials"`
	Action   string           `json:"action"`
	Category ActivityCategory `json:"category"`
	Source   string           `json:"source"`
}

// ActivityDayGroup groups audit entries under a day label.
type ActivityDayGroup struct {
	Label string `json:"label"`
	// Date is the group's calendar day as an ISO "2006-01-02" string. The
	// client filters and formats off this rather than parsing the display label.
	Date string        `json:"date"`
	Rows []ActivityRow `json:"rows"`
}

// ActivityLogPage is the payload for the Activity Log tab.
type ActivityLogPage struct {
	Categories []string           `json:"categories"`
	Users      []string           `json:"users"`
	Groups     []ActivityDayGroup `json:"groups"`
}

File: mozn-dashboard/web/types/activity.ts
/**
 * Activity Log API contract types. Mirror the JSON served by the Go backend
 * (internal/model/activity.go). Part of the shared types/ contract layer so the
 * fetch layer (lib/api) can reference them without reaching into features/. The
 * activity feature re-exports these from features/activity/types.
 */
import type { ActivityCategory } from "@/types/shared";

export type { ActivityCategory };

export interface ActivityRow {
  id: string;
  time: string;
  actor: string;
  initials: string;
  action: string;
  category: ActivityCategory;
  source: string;
}

export interface ActivityDayGroup {
  label: string;
  /** Calendar day as an ISO "YYYY-MM-DD" string; the client filters/formats off this. */
  date: string;
  rows: ActivityRow[];
}

/** Server-side pagination metadata (mirrors the backend `metadata` envelope). */
export interface ActivityLogMeta {
  page: number;
  pageSize: number;
  total: number;
  totalPages: number;
}

export interface ActivityLogPage {
  /** All selectable category facet values (not derived from the current page). */
  categories: string[];
  users: string[];
  groups: ActivityDayGroup[];
  /** Pagination cursor + totals for the current (server-filtered) query. */
  meta: ActivityLogMeta;
}

/** Full audit-log entry for the per-row detail view (GET /api/audit-logs/:id). */
export interface AuditLogDetail {
  id: string;
  action: string;
  resourceType: string;
  resourceId: string | null;
  status: string;
  statusCode: number;
  ipAddress: string;
  userAgent: string;
  durationMs: number;
  createdAt: string;
  requestPayload: unknown;
  responseError: string | null;
  details: unknown;
}
